drivers/src/mldsa87.rs

```diff
@@ -98,12 +153,46 @@ impl MlDsa87 {
     /// * `MlDsa87Signature` - Generated signature
     pub fn sign(
         &mut self,
-        _priv_key_in: &KeyReadArgs,
-        _pub_key: &MlDsa87PubKey,
-        _msg: &MlDsa87MsgScalar,
-        _trng: &mut Trng,
+        seed: &KeyReadArgs,
+        msg: &MlDsa87MsgScalar,
+        sign_rnd: &MlDsa87SignRndScalar,
+        trng: &mut Trng,
     ) -> CaliptraResult<MlDsa87Signature> {
-        Ok(MlDsa87Signature::default())
+        let mldsa = self.mldsa87.regs_mut();
+
+        // Wait for hardware ready
+        MlDsa87::wait(mldsa, || mldsa.status().read().ready())?;
+
+        // Clear the hardware before start
+        mldsa.ctrl().write(|w| w.zeroize(true));
+
+        // Copy seed from keyvault
+        KvAccess::copy_from_kv(*seed, mldsa.kv_rd_seed_status(), mldsa.kv_rd_seed_ctrl())
+            .map_err(|err| err.into_read_seed_err())?;
+
+        // Copy digest
+        KvAccess::copy_from_arr(msg, mldsa.msg())?;
+
+        // Sign RND, TODO do we want deterministic?
```

Comment:
We need deterministic, but we'll keep the API flexible. Caller will always send sign_rnd as zeroes.